collectors/ebpf.plugin/ebpf.c

```diff
@@ -374,7 +374,8 @@ static void ebpf_stop_threads(int sig)
 {
     int i;
     for (i = 0; ebpf_threads[i].name != NULL; i++) {
-        (void)netdata_thread_cancel(*ebpf_threads[i].thread);
+        if (ebpf_threads[i].thread && ebpf_threads[i].enabled == NETDATA_MAIN_THREAD_EXITED)
```

Comment:
Hm 🤔. You are using this condition in many places but you never assign `NETDATA_MAIN_THREAD_EXITED` to `ebpf_threads[i].enabled`. It just happened that `NETDATA_MAIN_THREAD_EXITED` is equal to `0`. What is your real logic with these conditions?

Reply:
@thiagoftsm why do we call `netdata_thread_cancel` for disabled threads?

Reply:
Hello guys, sorry for the delay, but you know the motive. :)
I rewrote the code related with these comments and I removed the calls for `netdata_thread_cancel`. See you on Tuesday!